Simplenote/src/main/java/com/automattic/simplenote/utils/AccountVerificationWatcher.java

```diff
@@ -18,57 +18,63 @@
  * have (and probably was) updated from another device.
  */
 public class AccountVerificationWatcher implements Bucket.OnNetworkChangeListener<Account> {
-    private enum Status {
-        UNKNOWN,
+    public enum Status {
+        SENT_EMAIL,
         UNVERIFIED,
         VERIFIED,
     }
 
+    public interface VerificationStateListener {
+        void onUpdate(Status status);
+    }
+
     private final Simplenote simplenote;
-    private Status currentStatus = Status.UNKNOWN;
+    private final List<VerificationStateListener> listeners = new ArrayList<>();
 
     public AccountVerificationWatcher(Simplenote simplenote) {
         this.simplenote = simplenote;
     }
 
-    public void clearStatus() {
-        currentStatus = Status.UNKNOWN;
+    public void addVerificationStateListener(VerificationStateListener listener) {
+        listeners.add(listener);
     }
```

Comment:
I should have thought about this yesterday, but _if_ you find this a bit overbearing with the listeners it's probably good enough to store just a single listener and pass it in when instantiating this thing. that would result in far less plumbing.

```java
private VerificationStateListener listener;

public AccountVerificationWatcher(Simplenote simplenote, VerificationStateListener listener) {
    this.simplenote = simplenote;
    this.listener = listener;
}
```

it seems unlikely we'll be adding more than one.